docs/build-on-linea/gas-fees.mdx

```diff
@@ -1,139 +1,108 @@
+</TabItem>
+<TabItem value="result" label="Result">
 
 ```bash
-curl https://linea-mainnet.infura.io/v3/your-api-key \
-  -X POST \
-  -H "Content-Type: application/json" \
-  -d '{"id": 1, "jsonrpc": "2.0", "method": "eth_feeHistory", "params": [5, "latest", [20]] }'
+{
+  "jsonrpc": "2.0",
+  "id": 53,
+  "result": {
+    "baseFeePerGas": "0x7",
+    "gasLimit": "0xcf08",
+    "priorityFeePerGas": "0x43a82a4"
+  }
+}
 ```
 
-# Example Code
+  </TabItem>
+</Tabs>
 
 :::note
-Remember to replace `maxFeePerGasFromConfig` with a value that fits your needs. A reasonable value for this is 10 Gwei.
+
+The result of the request returns hexadecimal equivalent integers of gas prices in wei. Convert the hexadecimal value into
+decimals to get the wei value. You can use any hexadecimal to decimal converter such as
+[RapidTables](https://www.rapidtables.com/convert/number/hex-to-decimal.html).
 
 :::
 
-```typescript
-import { ethers } from "ethers";
-
-export type Fees = {
-  maxFeePerGas: bigint;
-  maxPriorityFeePerGas?: bigint;
-};
-
-export type FeeHistory = {
-  oldestBlock: number;
-  reward: string[][];
-  baseFeePerGas: string[];
-  gasUsedRatio: number[];
-};
-
-export async function get1559Fees(
-  provider: ethers.JsonRpcProvider,
-  maxFeePerGasFromConfig: bigint,
-  percentile: number,
-): Promise<Fees> {
-  const { reward, baseFeePerGas }: FeeHistory = await provider.send(
-    "eth_feeHistory",
-    ["0x5", "latest", [percentile]],
-  );
-
-  const maxPriorityFeePerGas =
-    reward.reduce(
-      (acc: bigint, currentValue: string[]) => acc + BigInt(currentValue[0]),
-      0n,
-    ) / BigInt(reward.length);
-
-  if (maxPriorityFeePerGas && maxPriorityFeePerGas > maxFeePerGasFromConfig) {
-    throw new Error(
-      `Estimated miner tip of ${maxPriorityFeePerGas} exceeds configured max fee per gas of ${maxFeePerGasFromConfig}.`,
-    );
-  }
+### Formula
 
-  const maxFeePerGas =
-    BigInt(baseFeePerGas[baseFeePerGas.length - 1]) * 2n + maxPriorityFeePerGas;
-
-  if (maxFeePerGas > 0n && maxPriorityFeePerGas > 0n) {
-    return {
-      maxPriorityFeePerGas,
-      maxFeePerGas:
-        maxFeePerGas > maxFeePerGasFromConfig
-          ? maxFeePerGasFromConfig
-          : maxFeePerGas,
-    };
-  }
+The `linea_estimateGas` API calculates gas fees as follows:
 
-  return {
-    maxFeePerGas: maxFeePerGasFromConfig,
-  };
-}
-```
-
-# How do I check the gas price on Linea?
+- `baseFeePerGas` - Uses the Linea base fee which is static at 7 wei.
+- `gasLimit` - Uses the standard `eth_estimateGas` API calculation.
+- `priorityFeePerGas` - Uses the following formula:
 
-If you want to check the gas price on Linea, run the code below.
+    ```bash
+    priorityFeePerGas = L2_compressed_tx_size_in_bytes * L1_VERIFICATION_COST * L2_minimum_minable_gas_price * L1_L2_GAS_PRICE_RATIO * MINIMUM_MARGIN_ESTIMATE_GAS / (L2_tx_gas_used * L1_VERIFICATION_CAPACITY)
```

Comment:
@dfox-consensys do we really want to be as transparent as this?

Reply:
What about if we remove the defaults used and just describe the formula?

Reply:
@dfox-consensys, do you mean something like:

- `priorityFeePerGas` - Calculates the fee required to prioritize a transaction by considering factors such as the compressed transaction size, Layer 1 verification costs and capacity, gas price ratio between Layer 1 and Layer 2, the transaction's gas usage, the minimum minable gas price on Layer 2, and a minimum margin (for error) for gas price estimation."

Reply:
Yes, LGTM